flake.nix

```diff
@@ -21,12 +21,13 @@
 
         nixosSystem = args:
           import ./nixos/lib/eval-config.nix (
-            args // { inherit (self) lib; } // lib.optionalAttrs (! args?system) {
+            {
+              lib = final;
               # Allow system to be set modularly in nixpkgs.system.
               # We set it to null, to remove the "legacy" entrypoint's
               # non-hermetic default.
               system = null;
-            }
+            } // args
```

Comment:
Why would `lib = final` work? This would set it to the entire flakes output.

Reply:
`final` here is the final lib, it's the argument the `lib.extend` call, so it's essentially the same as `self.lib`.

Reply:
Ah right! I confused `self` with `final`